itou/siae_evaluations/models.py

```diff
@@ -0,0 +1,145 @@
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db import models
+from django.urls import reverse
+from django.utils import timezone
+
+from itou.institutions.models import Institution
+from itou.siae_evaluations import enums as evaluation_enums
+from itou.utils.emails import get_email_message
+
+
+def validate_institution(institution_id):
+    try:
+        institution = Institution.objects.get(pk=institution_id)
+    except Institution.DoesNotExist:
+        raise ValidationError("L'institution sélectionnée n'existe pas.")
+
+    if institution.kind != Institution.Kind.DDETS:
+        raise ValidationError(f"Sélectionnez une institution de type {Institution.Kind.DDETS}")
+
+
+def email_campaign_is_setup(emails, evaluated_period_start_at, evaluated_period_end_at, ratio_selection_end_at):
+    to = [settings.DEFAULT_FROM_EMAIL]
+    bcc = emails
+    context = {
+        "evaluated_period_start_at": evaluated_period_start_at,
+        "evaluated_period_end_at": evaluated_period_end_at,
+        "ratio_selection_end_at": ratio_selection_end_at,
+        "dashboard_url": f"{settings.ITOU_PROTOCOL}://{settings.ITOU_FQDN}{reverse('dashboard:index')}",
+    }
+    subject = "siae_evaluations/email/campaign_is_setup_subject.txt"
+    body = "siae_evaluations/email/campaign_is_setup_body.txt"
+    return get_email_message(to, context, subject, body, bcc=bcc)
+
+
+def create_campaigns(evaluated_period_start_at, evaluated_period_end_at, ratio_selection_end_at):
+    """
+    Create a campaign for each institution whose kind is DDETS.
+    This method is intented to be executed manually, until it will be automised.
+    """
+
+    name = (
+        f"contrôle pour la période du {evaluated_period_start_at.strftime('%d/%m/%Y')} "
+        f"au {evaluated_period_end_at.strftime('%d/%m/%Y')}"
+    )
+
+    institutions = Institution.objects.filter(kind=Institution.Kind.DDETS)
+
+    campaigns = [
+        EvaluationCampaign(
+            name=name,
+            evaluated_period_start_at=evaluated_period_start_at,
+            evaluated_period_end_at=evaluated_period_end_at,
+            institution=institution,
+        )
+        for institution in institutions
+    ]
+
+    evaluation_campaign = EvaluationCampaign.objects.bulk_create(campaigns)
```

Comment:
nit: un peu confusant, `bulk_create` va te retourner une liste de `created_campaigns`.
supernit: tu crees une liste avec `campaigns mais tu pourrais faire un itérateur ^^ on en parlera un jour si tu vois pas la différence

```python
# liste en mémoire
campaigns = [xxx for xxx in stuff]
EvaluationCampaign.objects.bulk_create(campaigns)

# itérateur, ne consomme pas de ram jusqu'au tout dernier moment (evaluation du queryset)
EvaluationCampaign.objects.bulk_create(xxx for stuff in stuff)  # pas de [] 
```